src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -404,13 +401,12 @@ void BrowseContext::OnBrowseRemove(const char * name, const char * type, const c
                     StringOrNullMarker(type), StringOrNullMarker(domain), interfaceId);
 
     VerifyOrReturn(name != nullptr);
-    VerifyOrReturn(IsLocalDomain(domain));
 
     services.erase(std::remove_if(services.begin(), services.end(),
                                   [name, type, interfaceId, domain](const auto & service) {
                                       return strcmp(name, service.first.mName) == 0 && type == GetFullType(&service.first) &&
                                           service.first.mInterface == chip::Inet::InterfaceId(interfaceId) &&
-                                          strcmp(domain, service.second.c_str()) == 0;
+                                          domain != nullptr && strcmp(domain, service.second.c_str()) == 0;
```

Comment:
How does `domain` get to be null here?

In any case, if it's null, shouldn't we treat it as empty string?  Otherwise if those entries got added with `std::string(nullptr)` as the domain, wouldn't we fail to remove them here?

All of which is to say that the simplest thing is to do something like:

```
std::string domain_str(domain);
```

before the `erase()` call, capture that by reference in the lambda, and use `service.second == domain_str` as part of the condition.

Reply:
And note that https://github.com/project-chip/connectedhomeip/pull/32779#discussion_r1544845831 is still not addressed.